### PR TITLE
Exception handling

### PR DESCRIPTION
**Pull request by abrahamdoosti**
service methods now return ResourceEntity and also GenericExceptionMapper is used to handle all exceptions thrown by the controller.
@belish

### CODE REVIEW

File: src/main/java/com/example/spring/boot/controller/StudentController.java [hunk 1]
```diff
@@ -23,27 +24,29 @@
...
-	public ResponseEntity<Student> updateStudent(@RequestBody Student student) {
+	@RequestMapping(value = "/{studentId}", method = RequestMethod.PUT, consumes = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Student> updateStudent(@RequestBody Student student,@PathVariable("studentId") int studentId) {
+		student.setId(studentId);
 		return studentService.updateStudent(student);
 	}
 	
 	@RequestMapping(value = "/{studentId}", method = RequestMethod.DELETE, consumes = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<Student> deleteStudent(@PathVariable("studentId") int id) {
+
```
Comment (belish) [inline src/main/java/com/example/spring/boot/controller/StudentController.java]: unnecessary space

File: src/main/java/com/example/spring/boot/controller/StudentController.java [hunk 2]
```diff
@@ -23,27 +24,29 @@
...
 	@RequestMapping(value = "/", method = RequestMethod.GET)
-	public List<Student> getAllStudents() {
-		return (List<Student>) studentService.getAllStudents();
+	public ResponseEntity<List<Student>> getAllStudents() throws StudentNotFoundException {
+		return  studentService.getAllStudents();
 	}
 
 	@RequestMapping(value = "/{studentId}", method = RequestMethod.GET)
-	public Student getStudent(@PathVariable("studentId") int id) {
+	public ResponseEntity<Student> getStudent(@PathVariable("studentId") int id) throws StudentNotFoundException {
```
Comment (belish) [inline src/main/java/com/example/spring/boot/controller/StudentController.java]: make parameter variable final

Comment (abrahamdoosti) [reply src/main/java/com/example/spring/boot/controller/StudentController.java]: final key word added.

File: src/main/java/com/example/spring/boot/controller/StudentController.java (hunk 1 above)
Comment (abrahamdoosti) [reply src/main/java/com/example/spring/boot/controller/StudentController.java]: space removed

**Comment by belish** [comment]
👍

**Comment by belish** [comment]
LGTM

**Pull request merged**